src/lib/cares_reply.c

```diff
@@ -0,0 +1,89 @@
+/* Copyright (C) 2021 by Kyle Evans
+ *
+ * Permission to use, copy, modify, and distribute this software and its
+ * documentation for any purpose and without fee is hereby granted, provided
+ * that the above copyright notice appear in all copies and that both that
+ * copyright notice and this permission notice appear in supporting
+ * documentation, and that the name of M.I.T. not be used in advertising or
+ * publicity pertaining to distribution of the software without specific,
+ * written prior permission.  M.I.T. makes no representations about the
+ * suitability of this software for any purpose.  It is provided "as is"
+ * without express or implied warranty.
+ */
+
+#include "ares_setup.h"
+#include "ares.h"
+#include "ares_private.h"
+#include "string.h"
+
+struct cares_srv_reply*
+cares_srv_reply_get_next(const struct cares_srv_reply* srv_reply)
+{
+  return srv_reply->next;
+}
+
+char* cares_srv_reply_get_host(const struct cares_srv_reply* srv_reply)
+{
+  char *newhost = NULL;
+  if ((newhost = malloc(strlen(srv_reply->host) + 1)) != NULL) {
+    strcpy(newhost, srv_reply->host);
+  }
+  return newhost;
+}
```

Comment:
I changed this to allocate a new `char*` and return it.  Just returning the raw pointer had the side effect of allowing direct modification of the host field of the struct using the pointer.  The downside to this method is that the caller will now have to both check for NULL and free the `char*`.  Let me know if there is a better way.

Also, what checks need to be done on srv_reply being passed here? Just check for NULL?  If it's not NULL then is it safe to assume I can `strcpy` without an overflow?

Reply:
I don't see a reason to do that, instead just make the return value   const char * and return the pointer ... that designates the caller isn't allowed to modify the return value.

Reply:
That makes sense, but in our case back in ares_parse_srv_reply, we will have to do this copy there to get host out of the cares_srv_reply struct and into the ares_srv_reply struct so that we can free the cares_srv_reply.  Is that okay then?